api/src/main/java/org/apache/iceberg/TrackingInfo.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import org.apache.iceberg.types.Types;
+
+/**
+ * Tracking information for a tracked file entry in a V4 manifest.
+ *
+ * <p>This groups the status, snapshot, and sequence number information for the entry. This enables
+ * accessing the fields for the entry and provides an isolated structure that can be modified.
+ */
+public interface TrackingInfo {
+  /** Status of an entry in a tracked file */
+  enum Status {
+    EXISTING(0),
+    ADDED(1),
+    DELETED(2);
+
+    private final int id;
+
+    Status(int id) {
+      this.id = id;
+    }
+
+    public int id() {
+      return id;
+    }
+  }
+
+  Types.NestedField STATUS =
+      Types.NestedField.required(0, "status", Types.IntegerType.get(), "Entry status");
+  Types.NestedField SNAPSHOT_ID =
+      Types.NestedField.optional(
+          1,
+          "snapshot_id",
+          Types.LongType.get(),
+          "Snapshot ID where the file was added, or deleted if status is 2. Inherited when null.");
```

Comment:
I think this should be `"Snapshot ID where the file was added or deleted/removed"`. No need to mention specific status codes or inheritance. People reading these docs should never see values without a snapshot ID because of inheritance so I don't think there is value in adding it here because this is primarily for metadata tables.